### PR TITLE
[LTR] Implement Boromir, Warden of the Tower

### PR DESCRIPTION
**Pull request by Susucre**

### CODE REVIEW

**Comment by Susucre** [comment]
[[Boromir, Warden of the Tower]]

File: Mage.Sets/src/mage/cards/b/BoromirWardenOfTheTower.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SOLDIER);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Vigilance
+        this.addAbility(VigilanceAbility.getInstance());
+
+        // Whenever an opponent casts a spell, if no mana was spent to cast it, counter that spell.
+        this.addAbility(new BoromirWardenOfTheTowerTriggeredAbility());
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/b/BoromirWardenOfTheTower.java:44]: Possible to factor out from this and Lavinia to common ability class?

Comment (Susucre) [reply Mage.Sets/src/mage/cards/b/BoromirWardenOfTheTower.java:44]: Sure,
my usual rule of thumb is to refactor when there are 3 instances of the same code piece, so did not bother making a class for that trigger.

File: Mage/src/main/java/mage/abilities/common/SpellCastOpponentNoManaSpentTriggeredAbility.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import mage.target.targetpointer.FixedTarget;
+
+/**
+ * @author Susucr
+ */
+public class SpellCastOpponentNoManaSpentTriggeredAbility extends TriggeredAbilityImpl {
+
+    public SpellCastOpponentNoManaSpentTriggeredAbility(Effect effect) {
+        super(Zone.BATTLEFIELD, effect, false);
+        this.setTriggerPhrase("Whenever an opponent casts a spell, if no mana was spent to cast it");
```
Comment (xenohedron) [inline Mage/src/main/java/mage/abilities/common/SpellCastOpponentNoManaSpentTriggeredAbility.java:18]: check text? I thought trigger phrase needed to end with `, `

Comment (Susucre) [reply Mage/src/main/java/mage/abilities/common/SpellCastOpponentNoManaSpentTriggeredAbility.java:18]: Thanks for the catch. I did indeed miss that.

**Review by xenohedron — APPROVED** [review]

**Pull request merged**